Source/CombatExtended/CombatExtended/Verbs/Verb_ShootCE.cs

```diff
@@ -94,10 +94,42 @@ protected override float SwayAmplitude
         // Whether our shooter is currently under suppressive fire
         private bool IsSuppressed => ShooterPawn?.TryGetComp<CompSuppressable>()?.isSuppressed ?? false;
 
+        //Checks if magazine-fed weapon needs to reload before able to shoot again (out-of-ammo state is automatically handled by TryStartReload)
+        private bool NeedsMagazineReload => CompAmmo != null && CompAmmo.HasMagazine && CompAmmo.CurMagCount == 0;
+
+        //Mag-less weapons need separate handling for out-of-ammo state, as TryStartReload does nothing for them
+        private bool IsMaglessWeaponOutOfAmmo => CompAmmo != null && !CompAmmo.HasMagazine && CompAmmo.UseAmmo && !CompAmmo.HasAmmo;
+
+        private bool IsAttacking => ShooterPawn?.CurJobDef == JobDefOf.AttackStatic || ShooterPawn?.stances.curStance is Stance_Warmup;
+
         #endregion
 
         #region Methods
 
+        public override bool Available()
+        {
+            if (!base.Available())
+            {
+                return false;
+            }
+
+            if (IsAttacking)
+            {
+                if (NeedsMagazineReload)
+                {
+                    CompAmmo?.TryStartReload();
+                    return false;
+                }
+                if (IsMaglessWeaponOutOfAmmo)
+                {
+                    CompAmmo?.DoOutOfAmmoAction();
```

Comment:
OutOfAmmoAction is already part of TryStartReload

Reply:
It is, but if you call TryStartReload on something like a bow, the function returns instantly because there isn't a magazine to reload, and the pawn will just sit there staring at their target, instead of switching to their next weapon like they should.

Reply:
That's a bug in TryStartReload then and should be fixed there, not here.